torchvision/transforms/functional_tensor.py

```diff
@@ -411,7 +413,28 @@ def pad(img: Tensor, padding: List[int], fill: int = 0, padding_mode: str = "con
         need_cast = True
         img = img.to(torch.float32)
 
-    img = torch_pad(img, p, mode=padding_mode, value=float(fill))
+    patch_padding = False
+    if isinstance(fill, (int, float)):
+        v = float(fill)
+    elif any([x != fill[0] for x in fill]):
+        patch_padding = True
+        v = 0.0
+    else:
+        v = fill[0]
```

Comment:
We only apply the patch if necessary. That is when `fill` is a list without constant values.